src/Forms/GridField/GridField.php

```diff
@@ -688,6 +689,13 @@ public function FieldHolder($properties = [])
         if ($total > 0) {
             $rows = [];
```

Comment:
Note this code is very close to another framework PR that's adding caching for the Version object (not just the Version number) https://github.com/silverstripe/silverstripe-framework/pull/11707/files#diff-c5907f790efc160f18abbccf8d6d3de8e13eff90e5b3a6119e1447da06f70c40R727